backend/de.metas.shipper.gateway.commons/src/main/java/de/metas/shipper/gateway/commons/model/CarrierShipperProduct.java

```diff
@@ -39,9 +41,16 @@ public class CarrierShipperProduct implements ShipperProduct
 	@Getter
 	String code;
 
+	String name;
+
 	public static CarrierShipperProduct ofCode(@NonNull final String code)
 	{
-		return new CarrierShipperProduct(code);
+		return ofCode(code, null);
```

Comment:
lound thinking: why not using the code as name too... so we can make name nonnull...